build.gradle

```diff
@@ -16,10 +16,50 @@
  */
 
 plugins {
-  id "org.nosphere.apache.rat" version "0.3.0"
+  id 'org.nosphere.apache.rat' version '0.3.0'
+  id 'org.ajoberstar.grgit' version '1.6.0'
 }
 
-apply plugin: "org.nosphere.apache.rat"
+apply plugin: 'org.ajoberstar.grgit'
+
+task clean << {
+  delete buildDir
+}
+
+def dockerCmd = "docker run -v '${projectDir}':/geode-site -p 3000:3000 -w /geode-site/website apachegeode/geode-site-build "
+
+task compile(type:Exec, dependsOn: rat) {
+  inputs.dir "${projectDir}/website"
+  outputs.dir "${buildDir}/content"
+  executable 'bash'
+  args "-c", dockerCmd + "nanoc compile"
+}
+
+task view(type:Exec, dependsOn: compile) {
+  executable 'bash'
+  args "-c", dockerCmd + "nanoc view"
+}
+
+task publish(dependsOn: compile) {
```

Comment:
Should the publish task also do the 'push' as well? When I think I am going to publish something, I would think it do all the steps.